test/integration/protocol_integration_test.cc

```diff
@@ -141,6 +141,8 @@ TEST_P(DownstreamProtocolIntegrationTest, RouterClusterNotFound503) {
 
 // Add a route which redirects HTTP to HTTPS, and verify Envoy sends a 301
 TEST_P(DownstreamProtocolIntegrationTest, RouterRedirect) {
```

Comment:
This test is equivalent to the local reply from the filter, down below. We should add byte checks here: https://github.com/envoyproxy/envoy/blob/1aba2de82c714e8c7573c4efc997cc112eba4acf/test/integration/redirect_integration_test.cc#L151 which tests internal redirect based on upstream response.

Reply:
Test added.